src/main/drivers/accgyro_fake.c

```diff
@@ -28,26 +50,55 @@
 #include "accgyro.h"
 #include "accgyro_fake.h"
 
-
 static int16_t fakeGyroADC[XYZ_AXIS_COUNT];
 
+#if defined(SIMULATOR_BUILD) && defined(SIMULATOR_MULTITHREAD)
```

Comment:
At first, I wanted to let it work with existing things,
so all the changes were as minimal as I can.
And I think not only acc and gyro but also IMU needs mutex.